motile/constraints/constraint.py

```diff
@@ -1,10 +1,16 @@
+from __future__ import annotations
+
 from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING
 
+if TYPE_CHECKING:
+    import ilpy
+    from motile.solver import Solver
```

Comment:
@funkey, let me know if you'd like more info on either the `__future__.annotations` pattern or `if typing.TYPE_CHECKING`

Reply:
Got the lesson offline, all good :)